### PR TITLE
Add some basic auto routes

### PR DESCRIPTION
**Pull request by ptavolette**
- Changed assignments for Positions 1-3 and Game pieces 1-4 (Reversed to match coordinate order)
- Used coordinates pulled from CAD to change and add coordinates to be more accurate
- Added initial routes for driving to game pieces, returning from those pieces, from all 3 positions

### CODE REVIEW

**Comment by MathNerd28** [comment]
Correct me if I’m wrong, but don’t the indices of the Driver Stations go in the original direction? I’m just worried it’s going to get confusing during competitions. It would be easier to say “We’re at station 3, so let’s start auto in position 3”.

File: src/main/java/frc/robot/AutoRoutes.java [hunk 1]
```diff
@@ -11,116 +11,232 @@
...
-            S3_P4_B, S3_P4_C, S3_P4_D, S3_P4_E, GAME_PIECE_4 };
-    public static final Pose2d[] PIECE_4_TO_START_3 = { GAME_PIECE_4, S3_P4_E,
-            S3_P4_D, S3_P4_C, S3_P4_B, S3_P4_A, START_POINT_3 };
-
-    static {
-        if (RobotBase.isSimulation()) {
-            new Thread(AutoRoutes::simulateTrajectories).start();
+        public static final double FIELD_LENGTH = 16.54;
+        public static final double FIELD_WIDTH  = 8.02;
+
```
Comment (MathNerd28) [inline src/main/java/frc/robot/AutoRoutes.java]: Please run the formatter on this file

File: src/main/java/frc/robot/AutoRoutes.java [hunk 2]
```diff
@@ -11,116 +11,232 @@
...
+        private static final Pose2d S1_P1_A = new Pose2d(1.49, 1.07, DEGREES_0);
+        private static final Pose2d S1_P1_B = new Pose2d(2.94, 0.92, DEGREES_0);
+        private static final Pose2d S1_P1_C = new Pose2d(4.83, 0.92, DEGREES_0);
+        private static final Pose2d S1_P1_D = new Pose2d(6.5, 0.92,
+                        DEGREES_180);
+        private static final Pose2d S1_P1_E = new Pose2d(6.82, 0.92,
+                        DEGREES_180);
+
+        public static final Pose2d[] START_1_TO_PIECE_1 = { S1_P1_A, S1_P1_B,
+                        S1_P1_C, S1_P1_D, S1_P1_E };
```
Comment (MathNerd28) [inline src/main/java/frc/robot/AutoRoutes.java]: These arrays must also include the start and end position, or else the robot will be very confused at the start!

Comment (MathNerd28) [reply src/main/java/frc/robot/AutoRoutes.java]: This applies to every array in this file

File: src/main/java/frc/robot/AutoRoutes.java [hunk 3]
```diff
@@ -11,116 +11,232 @@
...
+        // private static final Pose2d S3_P4_C = new Pose2d(5.7, 0.9,
+        // DEGREES_90);
+        // private static final Pose2d S3_P4_D = new Pose2d(6.0, 0.9,
+        // DEGREES_30);
+        // private static final Pose2d S3_P4_E = new Pose2d(6.3, 0.9,
+        // DEGREES_0);
+
+        // public static final Pose2d[] PIECE_4_TO_START_3 = { GAME_PIECE_4,
+        // S3_P4_E, S3_P4_D, S3_P4_C, S3_P4_B, S3_P4_A,
+        // START_POINT_3 };
```
Comment (MathNerd28) [inline src/main/java/frc/robot/AutoRoutes.java]: If these definitions are no longer valid, please delete them rather than just commenting them out.

**Review by MathNerd28 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/AutoRoutes.java (hunk 2 above)
Comment (Cameron-Williams1) [reply src/main/java/frc/robot/AutoRoutes.java]: We removed those as it looked like it in your code it was referencing these positions. "public static final Pose2d START_POINT_1 = new Pose2d(2.2, 4.7, DEGREES_180);
    public static final Pose2d START_POINT_3 = new Pose2d(2.2, 4.7, DEGREES_180);
    public static final Pose2d START_POINT_2 = new Pose2d(2.2, 2.75, DEGREES_180);
    public static final Pose2d START_POINT_3 = new Pose2d(2.2, 0.75, DEGREES_180);" 
As S1_P1_A is our starting position we kept it out

Comment (MathNerd28) [reply src/main/java/frc/robot/AutoRoutes.java]: That’s a duplication of the starting position; let’s give the common poses more clear names than “S1_P1_A”, and save those for intermediate poses we don’t care about.

**Review by MathNerd28 — APPROVED** [review]

**Pull request merged**